pkg/operator2/configobservation/routersecret/observe_router_secret.go

```diff
@@ -0,0 +1,90 @@
+package routersecret
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+
+	"github.com/openshift/cluster-authentication-operator/pkg/operator2/configobservation"
+)
+
+func ObserveRouterSecret(genericlisters configobserver.Listers, recorder events.Recorder, previouslyObservedConfig map[string]interface{}) (ret map[string]interface{}, _ []error) {
```

Comment:
just wondering why do we need it?  it didn't exist before, it's new, right?

Reply:
this did in fact exist before, if you look at the commit that adds it, it removes the respective config in `pkg/operator2/oauth` (as well as it moves the `routerSecretToSNI()` function from pkg/operator2/route.go).

My intention is to move all the configuration for the oauth-server eventually to observers, but that's still not happening today, and the reason is that that requires some heavy code changes, so I just picked the basic set of observers from https://github.com/openshift/cluster-authentication-operator/pull/222 that do not need that to make this PR more likely to go through the review process